core/src/data_sources/data_source.rs

```diff
@@ -915,9 +919,10 @@ impl DataSource {
                 let now = utils::now();
                 let chunk_len = chunk.len();
 
-                let _ = qdrant_client
-                    .upsert_points(self.qdrant_collection(), chunk, None)
-                    .await?;
+                try_join_all(qdrant_write_clients.iter().map(|qdrant_client| {
+                    qdrant_client.upsert_points(self.qdrant_collection(), chunk.clone(), None)
+                }))
```

Comment:
I'm fine going with that approach and not failing on shadow write logging only. I don't think the clone is costly. That just means we'll insert twice sequentially which much more costly than the clone but as you point it only during the migration.

Since that impacts only upsertion this is fine to add a bit of latency + this meam the happy path will remain "more unchnaged"

Will refactor the code accordingly.